server/lib/parties/party-service.ts

```diff
@@ -234,8 +239,38 @@ export default class PartyService {
   }
 
   leaveParty(partyId: string, userId: number, clientId: string) {
+    const party = this.parties.get(partyId)
+    if (!party) {
+      throw new PartyServiceError(PartyServiceErrorCode.PartyNotFound, 'Party not found')
+    }
+
+    if (!party.members.has(userId)) {
+      throw new PartyServiceError(PartyServiceErrorCode.UserNotInParty, 'User not in party')
+    }
+
     const clientSockets = this.getClientSockets(userId, clientId)
-    this.removeClientFromParty(clientSockets)
+    this.removeClientFromParty(clientSockets, party)
+  }
+
+  sendChatMessage(partyId: string, userId: number, message: string) {
+    const party = this.parties.get(partyId)
+    if (!party) {
+      throw new PartyServiceError(PartyServiceErrorCode.PartyNotFound, 'Party not found')
+    }
+
+    const user = party.members.get(userId)
+    if (!user) {
+      throw new PartyServiceError(PartyServiceErrorCode.UserNotInParty, 'User not in party')
+    }
+
+    const text = filterChatMessage(message)
+
+    this.publisher.publish(getPartyPath(partyId), {
+      type: 'chatMessage',
+      from: user,
+      time: Date.now(),
```

Comment:
As mentioned in the test above, don't retrieve the Date this way. This is an undeclared dependency and causes headaches for testing